### ISSUE TITLE
extend AnalogSignalArray min, max, mean, std, ... to operate across signals, and not just within epochs

### ISSUE THREAD

**Issue by eackermann** [eackermann/nelpy#127]
Similar to np.mean(), np.max(), etc. that takes an axis argument, we should support that, too.

What if, for example, we want to get the average over 4 LFP channels?

**Comment by eackermann** [comment]
If I had asa.mean(axis=1), then I could use _that_ one to plot my LFP instead of picking out a particular channel, as mentioned in #126

**Comment by shayokdutta** [comment]
I can easily add in an axis argument but just to make sure you understand...
Current functionality is with axis=0 and it returns the mean of each of the signals within ASA. Changing it to axis=1 would return the mean across all data points with each associated signal. This is better illustrated by the example below

![screenshot from 2017-03-04 22-50-56](https://cloud.githubusercontent.com/assets/6528643/23584656/2a267140-012d-11e7-8a1e-603d361e5aa8.png)

**Comment by eackermann** [comment]
Yes, that's what I was asking for, but of course, it might make sense to return a new AnalogSignalArray.

This points to an unsolved problem, actually. How should we store intermediate (and sometimes final) results? 

The AnalogSignalArray, SpikeTrainArray, EpochArray, etc. are very useful and flexible objects, but we often need something _different_, which is more difficult (at least for now) to define.

Case in point: if we do ASA.mean() it makes sense to return an np.array with the mean values for each of the signals. But if we do ASA.mean(axis=1), then it is perhaps more natural to want an AnalogSignalArray back? 

One solution (if it returns an np.array) is to simply build it from scratch, by doing e.g.
`
newasa = nel.AnalogSignalArray(ydata=asa.mean(axis=1), tdata=asa.tdata, support = asa.support, fs=asa.fs)
`
but even this won't actually work very well, as soon as we work with propagating additional metadata (which has not been implemented in ASA yet).

But also, it kind of looks ugly, don't you think? I would hope to be able to simply do something like
`asa.mean(axis=1)` to get a new ASA out, just like currently I can do `spktrn.flatten()` to return a new SpikeTrainArray object with all the metadata and attributes. but where all the spikeatrains are collapsed across units.

How can we do this elegantly and naturally? It's no good if asa.mean(axis=0) returns one type of object, and asa.mean(axis=1) returns a _different_ type of object, right? (Although I think that _is_ the desired behavior in 99% of cases).

One possibility might be to not use mean(axis=1) but to have a different method to do what we want (but what should we name this and other similar methods?) and another possibility is to override casting to numpy arrays, so that you can do, e.g.

`myarray = np.array(asa.mean(axis=1)) which would just cut out the tdata and ydata, leaving everything else out, but this might be complicated, and I'd rather access the underlying attributes directly when I need them.

So in sum, this might warrant some more discussion.

I'm having similar issues with returning loosely-defined results from my HMMs, which is at least part of why it's difficult to generate that plot with the LFP and the position and the states all lined up, since the results that you return invariably lose some info along the way, just like asa.mean(axis=1) would lose the tdata and the support epoch data, and fs, and everything else, if we only return the equivalent of `np.mean(asa.ydata, axis=1)`.

What are your thoughts?
